package.json

```diff
@@ -45,13 +45,14 @@
     "@types/eslint": "^7.2.8",
     "@types/estree": "^0.0.47",
     "@types/node": "^14.14.37",
-    "@typescript-eslint/parser": "^2.34.0",
+    "@typescript-eslint/parser": "^3.10.1",
```

Comment:
this seems like it would affect our ability to test on v2 of the parser. any thoughts/concerns?

Reply:
I've been looking at this [changelog of `2.34.0` -> `3.0.0`](https://github.com/typescript-eslint/typescript-eslint/commits/v3.0.0/packages/parser) and don't think there's anything special in v2 that is being missing after upgrading to v3. Most important thing is the node v8 support being dropped out but I think that's it. But I must say I'm not expert when it comes to AST parsers. 🙃

Reply:
That means we wouldn't be able to use it on node 4 and 6 tests :-/

Reply:
That's definitely a drawback. Not sure how to resolve this. 😕 

A bit off-topic but just to clarify - how was the `v2.34.0` be able to run on node 4? It [requires node v8 + eslint v5](https://github.com/typescript-eslint/typescript-eslint/blob/v2.34.0/packages/parser/package.json#L12-L14). ESLint v5 [requires node 6](https://github.com/eslint/eslint/blob/5.x/package.json#L125-L127). Am I reading these requirements correctly?

Reply:
actually you're right; the tests are set up to skip the TS tests on node < 5 (or maybe <= 5).

(this project supports down to eslint 3, which requires node 4)

Reply:
Alright so previously with `v2.34.0` the TS tests have been run for `eslint@^5` + `node@^8`. 
Now the `v3` supports `eslint@^5` and `node@^10`. The only thing that is lost is being able to test the typescript parser against node v8. ESLint version is still the same. That doesn't sound too bad, right?

Reply:
I think we can fix this by depending on `^2.34.0 || ^3.10.1`, and in the CI config, manually installing v2 when it's node < 10.

Reply:
Now using `"@typescript-eslint/parser": "^2.34.0 || ^3.10.1"` and CI seems to be passing. Does it still require changes to CI workflow configurations? 

Should the version be forced here?

https://github.com/yannickcr/eslint-plugin-react/blob/f233eb77c5c55662287991a914f3a7e2c4172db0/.github/workflows/node-4%2B.yml#L65

Reply:
Yes, I think that's precisely the solution that's needed :-)

Reply:
CI will now resolve version based on `matrix.node`: `"@typescript-eslint/parser@${{ matrix.node-version >= 10 && '3' || '2' }}"`.
The "ternary" syntax here is a bit weird but it works.

Logs from jobs:
- `latest majors (10, 7)`: `* AFTER_INSTALL: npm install --no-save "eslint@7" "@typescript-eslint/parser@3"`
- `latest majors (9, 6)`: `* AFTER_INSTALL: npm install --no-save "eslint@6" "@typescript-eslint/parser@2"`